Reference for...of statement before usage in example

daverayment opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

The 'Showing file(s) size' example includes a `for...of` statement to iterate through the file array. However, this loop construct has not been introduced previously on the page; instead a classic `for` loop with an iteration variable is shown as a method of iterating through `FileList` entries.

<!-- ✍️ Summarize your changes in one or two sentences -->

To prevent potential confusion, this change mentions the `for...of` statement and gives a snippet of how to use it before it's referenced in the example.

<!-- ❓ Why are you making these changes and how do they help readers? -->

This change prevents potential confusion caused by the current documentation using a different loop statement for the example than was used immediately prior.

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

daverayment committed:
Reference for...of statement before usage in example

The 'Showing file(s) size' example includes a for...of statement to iterate through the file array. However, this has not been introduced previously in the page (instead a for loop with an iteration variable was used just above). To prevent potential confusion, this change mentions the for...of statement and gives a snippet of how to use it before it's referenced.

Alternatively, I'd suggest changing the 'Showing file(s) size' example to use a for loop with an iteration variable for consistency.

wbamberg reviewed (commented):
Thank you @daverayment , I agree that this us a helpful change. I expect what happened is that the example used to use `for(;;)`, but then someone updated it and didn't notice the preamble.

If it were me I'd just remove the first version entirely, it's not relevant for this example and we should prefer `for...of` here (we have a [guideline](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#loop_initialization) about this).

That said, this PR is an improvement as-is, so if you don't want to make the change I'm happy to merge it. But please let me know :).

daverayment committed:
Remove file iteration examples

Readers will be familiar with looping through arrays, so the `for (;;)` and `for...of` snippets are not required.

daverayment commented:
Thank you for the kind review, @wbamberg . I agree that the section is as clear without the iteration examples, so have removed my proposed edit and the `for (;;)` loop example in a new commit.

wbamberg reviewed (approved):
👍 thank you @daverayment , and sorry to be so slow getting to your PR!
